pastis/plotting.py

```diff
@@ -994,3 +995,145 @@ def natural_keys(text):
+    if save:
+        fname = f'stat_1d_mus_{c_target}'
+        plt.savefig(os.path.join(out_dir, '.'.join([fname, 'pdf'])))
+    else:
+        plt.show()
+
+
+def plot_zernike_mus(mus, nmodes, nsegments, c_target, out_dir, save=False):
+    """
+    Plot localized zernike modal constraints plot for individual segment.
+
+    Parameters
+    ----------
+    mus : ndarray
+        list of standard deviations for each segment in nm
+    nmodes : int
+        number of thermal modes
+    nsegments :  int
+        number of segments
+    c_target : scalar
+        target contrast
+    out_dir : str
+        path to save the plot, if save=True
+    save : bool
+        whether to save the plot, if False, it shows the plot
+    """
+
+    coeffs_table = pastis.util.sort_1d_mus(mus, nmodes, nsegments)
+
+    plt.figure(figsize=(10, 10))
+    plt.title("Modal constraints to achieve a dark hole contrast of %.2e" % c_target, fontsize=20)
+    plt.ylabel("Weight per segment (in units of pm)", fontsize=15)
+    plt.xlabel("Segment Number", fontsize=20)
+    plt.tick_params(top=True, bottom=True, left=True, right=True,
+                    labelleft=True, labelbottom=True, labelsize=20)
+
+    for i in range(nmodes):
+        plt.plot(coeffs_table[i]*1e3, label='Zernike mode: %s' % i)
+
+    plt.grid()
+    plt.legend(fontsize=15)
+    plt.tight_layout()
+
+    if save:
+        fname = f'stat_1d_mus_{c_target}'
+        plt.savefig(os.path.join(out_dir, '.'.join([fname, 'pdf'])))
+    else:
+        plt.show()
+
+
+def plot_multimode_mus_surface_map(tel, mus, num_modes, num_actuators, c_target, data_dir,
+                                   mirror, cmin, cmax, save=False):
+    """
+    Creates surface deformation tolerance maps for localized wavefront aberrations.
+
+    Parameters:
+    tel : class instance of the simulator for "instrument"
+    mus : 1d array
+        list of tolerances for individual segments in units of nm
+    num_modes : int
+        the total number of local modes used to poke a segment.
+        For a harris segment mirror  it is 5 (taking only thermal basis map).
+        For a segmented mirror, it represents the total number of local zernike modes.
+    num_actuators : int
+        total number of dm actuators
+    c_target : float
+        desired dark hole contrast for which the tolerancing is done
+    data_dir :  str
+        path to save the plot, if save=True
+    cmin : float
+        minimum value for colorbar plot
+    cmax : float
+        minimum value for colorbar plot
+    save : bool
+        whether to save the plot
+    """
+    nm_aber = CONFIG_PASTIS.getfloat('LUVOIR', 'calibration_aberration') * 1e-9
+
+    coeffs_mumaps = pastis.util.calculate_mu_maps(mus, num_modes, num_actuators, tel.nseg)
+
+    mu_maps = []
+    for qq in range(num_modes):
+        coeffs = coeffs_mumaps[qq] # in units of nm
+        if mirror == 'harris_sm':
+            tel.harris_sm.actuators = coeffs * nm_aber / 2 # in units of m
```

Comment:
@ananyasahoo1904 why do you multiply by `nm_aber` here and below when applying the coefficients to the mirror?

Reply:
@ivalaginja The original mus generated from the PASTIS matrix were in units of poke aberration. Here poke aberration was 1nm, so units were of mus were 1 nm. I wanted to convert this to the SI units,  just to be on safer side and double checking.

Reply:
If `nm_aber` is 1 nm, multiplying by it doesn't do anything. All you are doing here is actually doing a unit conversion with the factor `1e-9` in the line where you read `nm_aber` from the configfile. If `nm_aber` were not 1 nm though, you would be scaling the actuator coefficients by an arbitrary number that was intended to calculate the matrix. The matrix itself is already normalized by this, and the result is always saved in units of nm.

This means that the `nm_aber` multiplication in the actuator command needs to be replaced just with the correct unit converstion to make it be meters, `1e-9` in this case here. I did this over in #143.